demos-europe/demosplan-core
# F t30254 autoload routes

Ticket:
https://yaits.demos-deutschland.de/T30254

Description:
implemented the use of AnnotationDirectoryLoader to load addon annotations of enabled addons automatically
if provided under ```AddonInfo::getInstallPath()./src/Controller```

implemented a special treatment for SlicingTagging because it lives within demosPipe at the moment
but is structured like a separate addon.
Therefore a hardcoded path to find its controllers was needed without it throwing an exception if the path does not
exist - which will be the case for all projects where demosPipe is disabled/notInstalled.

### Linked PR's
https://github.com/demos-europe/demosplan-addon-demospipes/pull/37
https://github.com/demos-europe/demosplan-core/pull/1002

### How to review/test
check if you can run <project> dp:c:c
check if the route api/1.0/AnnotatedStatementPdfPercentageDistribution/ exists (ewm->login->übersicht)
